scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -1286,6 +1312,67 @@ def _create_build_dir_link(self, links_dir_path, instance):
 
         self.link_dir_counter += 1
 
+    def handle_additional_scripts(
+        self, platform_name: str, testsuite: TestInstance
+    ) -> bool:
+        logger.debug(testsuite.testsuite.id)
+        matched_scripting = self.scripting.get_matched_scripting(
+            testsuite.testsuite.id, platform_name
+        )
+        if matched_scripting:
+            # Define a function to validate
+            # if the platform is supported by the matched scripting
+            def validate_boards(platform_scope, platform_from_yaml):
+                return any(board in platform_scope for board in platform_from_yaml)
+
+            # Define the types of scripts we are interested in as a set
+            script_types = {
+                "pre_script": "pre_script_timeout",
+                "post_flash_script": "post_flash_timeout",
+                "post_script": "post_script_timeout",
+            }
+
+            # Iterate over all DUTs to set the appropriate scripts
+            # if they match the platform and are supported
+            for dut in self.env.hwm.duts:
+                # Check if the platform matches and if the platform
+                # is supported by the matched scripting
+                if dut.platform in platform_name and validate_boards(
+                    platform_name, matched_scripting.platforms
+                ):
```

Comment:
```suggestion
                if dut.platform in platform_name:
```

Reply:
I was asked to enhance security to ensure that users are fully aware when using the --scripting-list parameter. To address this, I have introduced parameter  `override_scripts` which manages the redeclaration of provided scripts. Your concerns about _matches_elemet function and duplicates, lets clear how its work based on examples :

1.  `example_script.sh` is provided from command line via parameter `-pre-script ./example_script.sh` or from map.file
     **_`_matches_elemet `is not called, script executed : `example_script.sh`_**

2.   `example_script.sh` is provided from command line via parameter `-pre-script ./example_script.sh` or from map.file other_script.sh is provided with --scripting-list
```
       - scenarios:
          - <EXAMPLE_SCENARIO>
          platforms:
          - <EXAMPLE_PLATFORM>
          pre_script:
              path: /example_script.sh
          comment: >
            <COMMENT>
```

**_there will be no script executed for scenario " EXAMPLE_SCENARIO " on platform "EXAMPLE_PLATFORM", there will be log message `pre_script will not be overridden on EXAMPLE_PLATFORM` and none script will be executed, on other scenarios and boards `-pre-script ./example_script.sh` will be executed_**

3.  `example_script.sh` is provided from command line via parameter `-pre-script ./example_script.sh` or from map.file
       other_script.sh is provided with --scripting-list
```
       - scenarios:
          - <EXAMPLE_SCENARIO>
          platforms:
          - <EXAMPLE_PLATFORM>
          pre_script:
              path: /example_script.sh
              **override_script: true**
          comment: >
            <COMMENT>
```
 **_script `/example_script.sh` will be called only for scenario EXAMPLE_SCENARIO on platform EXAMPLE_PLATFORM, other configurations will be executed with `-pre-script ./example_script.sh`_**

4. only --scripting-list is provided with duplicate without override_script : 
```
       - scenarios:
          - <EXAMPLE_SCENARIO>
          platforms:
          - <EXAMPLE_PLATFORM>
          pre_script:
              path: /example_script1.sh
          comment: >
            <COMMENT>
                   - scenarios:
          - <EXAMPLE_SCENARIO>
          platforms:
          - <EXAMPLE_PLATFORM>
          pre_script:
              path: /example_script2.sh
          comment: >
            <COMMENT>
            
``` 

_**there will be log message that none of this scripts will be executed**_

5. only --scripting-list is provieded with duplicated override_script :
```
       - scenarios:
          - <EXAMPLE_SCENARIO>
          platforms:
          - <EXAMPLE_PLATFORM>
          pre_script:
              path: /example_script1.sh
              **override_script: true**
          comment: >
            <COMMENT>
                   - scenarios:
          - <EXAMPLE_SCENARIO>
          platforms:
          - <EXAMPLE_PLATFORM>
          pre_script:
              path: /example_script2.sh
             **override_script: true**
          comment: >
            <COMMENT>
            
```
**_Execution will be stopped with log message_**

6. only --scripting-list is provieded with duplicated records (only one override_script is declared) :
```
       - scenarios:
          - <EXAMPLE_SCENARIO>
          platforms:
          - <EXAMPLE_PLATFORM>
          pre_script:
              path: /example_script1.sh
          comment: >
            <COMMENT>
                   - scenarios:
          - <EXAMPLE_SCENARIO>
          platforms:
          - <EXAMPLE_PLATFORM>
          pre_script:
              path: /example_script2.sh
             **override_script: true**
          comment: >
            <COMMENT>
            
```

**_/example_script2.sh will be executed for scenario EXAMPLE_SCENARIO for platform_**


Handling duplicates works, i see that you aware about Input sources for scenarios and platforms could be not controlled cause of regexp, but data for `_matches_script` are loaded and validated form schema `scripting_schema` using the `scl.yaml_load_verify` function - this ensures that the input conforms to known structure,` load_from_yaml()` function - watch for invalid regex pattern is encountered if yes, exception will be raised.